### PR TITLE
Consolidate projects into one src and one test.

### PR DESCRIPTION
**Pull request by adamhathcock**
Covering https://github.com/frankhommers/Hangfire.PostgreSql/issues/60 and removing nuspec as mentioned on https://github.com/frankhommers/Hangfire.PostgreSql/pull/59

Need help with psake as powershell is gibberish to me :/

### CODE REVIEW

**Comment by adamhathcock** [comment]
Maybe this isn't that usable without fixing me psake (though, I'm hoping someone else will fix) but you can just use `dotnet` cli commands instead of nuget or whatever.

If it helps, I use Cake and this is a sample build I use: https://github.com/Visibilityltd/Visibility.AspNetCore.DataProtection.DistributedCache/blob/master/build.cake

psake is hard for me being on OS X.

File: src/Hangfire.PostgreSql/project.json [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+        "Hangfire.Core": "1.6.7",
+        "Newtonsoft.Json": "9.0.1",
+        "Npgsql": "3.1.9"
+    },
+    "scripts": {
+        "postcompile": [
+            "dotnet pack --no-build --configuration %compile:Configuration%"
+        ]
+    },
+    "frameworks": {
```
Comment (adamhathcock) [inline src/Hangfire.PostgreSql/project.json]: does both frameworks now

**Comment by adamhathcock** [comment]
Can run in the directory:
`dotnet restore`
`dotnet build src/**/project.json`

**Comment by jenyayel** [comment]
Wow, massive change, but good one. I would differently switch to Cake instead of psake. I can give a hand, but it will take a few days (busy period). 

Also worth adding CI into [Status checks](https://developer.github.com/v3/repos/statuses/), so tests will be executed once PR opened, but it requires [owners permissions](https://github.com/frankhommers/Hangfire.PostgreSql/settings/branches/master).

**Comment by adamhathcock** [comment]
I can help with Cake.

I use it for Sharpcompress as well as using Appveyor too:
https://github.com/adamhathcock/sharpcompress/blob/master/appveyor.yml

https://github.com/adamhathcock/sharpcompress/blob/master/build.cake

**Comment by vytautask** [comment]
@adamhathcock that would be nice.

**Comment by vytautask** [comment]
@adamhathcock does both (.net and .net core) tests pass?

**Comment by adamhathcock** [comment]
Tests pass!

Removed psake and put in Cake:
`./build.ps1` will do Tests.
`./build.ps1 -t Pack` will make a nuget package

Need to test out appveyor config

File: build.cake [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+  .Does(() =>
+{
+    DotNetCoreRestore("src/\" \"tests/");
+});
+
+Task("Build")
+    .IsDependentOn("Restore")
+  .Does(() =>
+{
+    DotNetCoreBuild("src/**/project.json\" \"tests/**/project.json");
```
Comment (jenyayel) [inline build.cake]: Won't build on *nix (travis) - there is no full framework there. You need something like:
```
	var projects = GetFiles("./src/**/project.json");
	foreach(var projectFile in projects)
	{
		var frameworks = ParseJsonFromFile(projectFile)
			.GetValue("frameworks")
			.Children<JProperty>()
			.Select(j => j.Name);

		foreach(var framework in frameworks)
		{
			if (!IsRunningOnWindows() && framework == "net452")
				continue;
			
			Information("Framework version {0}", framework);
			var settings = new DotNetCoreBuildSettings
			{
				Framework = framework
			};
			DotNetCoreBuild(projectFile.FullPath, settings);
		}			
	}
```

File: build.cake [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+});
+
+Task("Pack")
+    .IsDependentOn("Test")
+  .Does(() =>
+{
+    var settings = new DotNetCorePackSettings
+    {
+        Configuration = "Release",
+        OutputDirectory = "publish/"
```
Comment (jenyayel) [inline build.cake]: won't hurt `NoBuild = true`

**Comment by adamhathcock** [comment]
Updated for @jenyayel

**Comment by ericgreenmix** [comment]
Loving this change. Also, glad it switched to netstandard instead of netcoreapp. Will be massively helpful

**Comment by ericgreenmix** [comment]
Any update on getting this pulled in?

**Pull request merged**